internal/tools/aws/rds.go

```diff
@@ -0,0 +1,193 @@
+		DBClusterIdentifier:   aws.String(awsID),
+		DatabaseName:          aws.String("mattermost"),
+		EngineMode:            aws.String("provisioned"),
+		Engine:                aws.String("aurora-mysql"),
+		EngineVersion:         aws.String("5.7"),
+		MasterUserPassword:    aws.String(password),
+		MasterUsername:        aws.String(username),
+		Port:                  aws.Int64(3306),
+		StorageEncrypted:      aws.Bool(false),
+		DBSubnetGroupName:     aws.String(DefaultDBSubnetGroupName),
+		VpcSecurityGroupIds:   aws.StringSlice(dbSecurityGroupIDs),
+	}
+
+	_, err = svc.CreateDBCluster(input)
+	if err != nil {
+		return err
+	}
+
+	logger.WithField("db-cluster-name", awsID).Debug("AWS DB cluster created")
+
+	return nil
+}
+
+func (a *Client) rdsEnsureDBClusterInstanceCreated(awsID, instanceName string, logger log.FieldLogger) error {
+	svc := rds.New(session.New(), &aws.Config{
+		Region: aws.String(DefaultAWSRegion),
+	})
+
+	_, err := svc.DescribeDBInstances(&rds.DescribeDBInstancesInput{
+		DBInstanceIdentifier: aws.String(instanceName),
+	})
+	if err == nil {
+		logger.WithField("db-instance-name", instanceName).Debug("AWS DB instance already created")
+
+		return nil
+	}
+
+	_, err = svc.CreateDBInstance(&rds.CreateDBInstanceInput{
+		DBClusterIdentifier:  aws.String(awsID),
+		DBInstanceIdentifier: aws.String(instanceName),
+		DBInstanceClass:      aws.String("db.r5.large"),
+		Engine:               aws.String("aurora-mysql"),
+		PubliclyAccessible:   aws.Bool(false),
+	})
+	if err != nil {
+		return err
+	}
+
+	logger.WithField("db-instance-name", instanceName).Debug("AWS DB instance created")
+
+	return nil
+}
+
+func rdsGetDBCluster(awsID string, logger log.FieldLogger) (*rds.DBCluster, error) {
+	svc := rds.New(session.New(), &aws.Config{
+		Region: aws.String(DefaultAWSRegion),
+	})
+
+	result, err := svc.DescribeDBClusters(&rds.DescribeDBClustersInput{
+		DBClusterIdentifier: aws.String(awsID),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(result.DBClusters) != 1 {
+		return nil, fmt.Errorf("expected 1 DB cluster, but got %d", len(result.DBClusters))
+	}
+
+	return result.DBClusters[0], nil
+}
+
+func (a *Client) rdsEnsureDBClusterDeleted(awsID string, logger log.FieldLogger) error {
+	svc := rds.New(session.New(), &aws.Config{
+		Region: aws.String(DefaultAWSRegion),
+	})
+
+	result, err := svc.DescribeDBClusters(&rds.DescribeDBClustersInput{
+		DBClusterIdentifier: aws.String(awsID),
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == rds.ErrCodeDBClusterNotFoundFault {
+				logger.WithField("db-cluster-name", awsID).Warn("DBCluster could not be found; assuming already deleted")
+
+				return nil
+			}
+		}
+		return err
+	}
+
+	if len(result.DBClusters) != 1 {
+		return fmt.Errorf("expected 1 DB cluster, but got %d", len(result.DBClusters))
+	}
+
+	for _, instance := range result.DBClusters[0].DBClusterMembers {
+		_, err = svc.DeleteDBInstance(&rds.DeleteDBInstanceInput{
```

Comment:
Hmm, now I'm noticing that there are a number of places, like this, where we don't use the return value but also are not using the `if _, err = thing(); err != nil {` pattern -- is this a style choice that I'm ignorant of?

Reply:
Yeah, this is a style choice for the cloud repo. I personally prefer it, but other initial contributors to the repo also used the same style.

Reply:
Sounds good, I'm more than happy to adhere to project style.